pkg/util/httputil/client.go

```diff
@@ -78,6 +98,37 @@ func (c *Client) Get(ctx context.Context, url string) (resp *http.Response, err
 	return c.Do(req)
 }
 
+// Put does like http.Client.Put but uses the provided context and obeys its cancellation.
+func (c *Client) Put(
+	ctx context.Context, url string, h *RequestHeaders, body io.Reader,
+) (resp *http.Response, err error) {
+	req, err := http.NewRequestWithContext(ctx, "PUT", url, body)
+	if err != nil {
+		return nil, err
+	}
+	if h.ContentType != "" {
+		req.Header.Set("Content-Type", h.ContentType)
+	}
+	if h.Authorization != "" {
+		req.Header.Set("Authorization", h.Authorization)
+	}
+	return c.Do(req)
+}
+
+// Delete does like http.Client.Delete but uses the provided context and obeys its cancellation.
+func (c *Client) Delete(
+	ctx context.Context, url string, h *RequestHeaders,
+) (resp *http.Response, err error) {
+	req, err := http.NewRequestWithContext(ctx, "DELETE", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	if h.Authorization != "" {
```

Comment:
Same as above; either allow `nil` or state the precondition s.t. the caller doesn't make the wrong assumption.